File: C++/WilczeJagody.cpp
#include <iostream>
#include <windows.h>
#include "WilczeJagody.h"

WilczeJagody::WilczeJagody(int x, int y, Swiat* swiat) : Roslina(x, y, 99, 0, IDWILCZEJAGODY, swiat)
{
}
WilczeJagody::WilczeJagody(int x, int y, bool dz, Swiat* swiat) : Roslina(x, y, dz, 99, 0, IDWILCZEJAGODY, swiat)
{
}

void WilczeJagody::rysowanie()
{
	HANDLE col;
	col = GetStdHandle(STD_OUTPUT_HANDLE);
	SetConsoleTextAttribute(col, KOLORWILCZEJAGODY);
	std::cout << "J";
}

std::string WilczeJagody::nazwa()
{
	return "J";
}

void WilczeJagody::stworz_nowe_Roslina(int x, int y, Swiat* sw)
{
	Organizm* org = new WilczeJagody(x, y, true, sw);
}

int WilczeJagody::SzansaNaRozsiew()
{
	return 100;
}

WilczeJagody::~WilczeJagody()
{
}

File: Python/WilczeJagody.py
from Roslina import Roslina
from Organizm import Organizm

class WilczeJagody(Roslina):
    def __init__(self, x, y, dz, swiat):
        super().__init__(x, y, dz, 99, 0, swiat.IDWILCZEJAGODY, swiat)

    def rysowanie(self):
        return "#04167a"

    def nazwa(self):
        return "Wilcze Jagody"

    def stworz_nowe_roslina(self, x, y, sw):
        org = WilczeJagody(x, y, True, sw)

    def szansa_na_rozsiew(self):
        return 100
